1-js/03-code-quality/06-polyfills/article.md

```diff
@@ -1,92 +1,92 @@
-    // Math.ceil and Math.floor exist even in ancient JavaScript engines
-    // they are covered later in the tutorial
+if (!Math.trunc) {
+  // اگر چنین فانکشنی وجود ندارد
+  // آن زا پیاده‌سازی کن
+  Math.trunc = function (number) {
+    // Math.ceil و Math.floor جتی در موتورهای قدیمی جاوااسکریپت هم حضور دارند
+    // و در همین tutorial جلوتر آموزش داده می‌شوند
     return number < 0 ? Math.ceil(number) : Math.floor(number);
   };
 }
 ```
 
-JavaScript is a highly dynamic language, scripts may add/modify any functions, even including built-in ones. 
+جاوااسکریپت یک زبان به‌شدت داینامیک است. اسکریپت‌ها می‌توانند هر فانکشنی را تغییر دهند یا اضافه کنند. حتی فانکشن‌های built-in.
 
-Two interesting libraries of polyfills are:
-- [core js](https://github.com/zloirock/core-js) that supports a lot, allows to include only needed features.
-- [polyfill.io](http://polyfill.io) service that provides a script with polyfills, depending on the features and user's browser.
+دو کتابخانه جالب polyfills:
 
+- [core js](https://github.com/zloirock/core-js) که از چیزهای زیادی پشتیبانی می‌کند و اجازه می‌دهد فقط فیچرهای مورد نیاز خود را اضافه کنید.
+- [polyfill.io](http://polyfill.io) سرویسی که یک اسکریپت با polyfills ارائه می‌دهد. بسته به فیچرها و مرورگر کاربر.
 
-## Summary
+## خلاصه
 
-In this chapter we'd like to motivate you to study modern and even "bleeding-edge" language features, even if they aren't yet well-supported by JavaScript engines.
+در این فصل ما می‌خواهیم به شما انگیزه بدهیم تا بروید و فیچرهای جدید زبان را یاد بگیرید. حتی اگر هنوز توسط موتورهای جاوااسکریپت به‌خوبی پشتیبانی نمی‌شوند.
 
-Just don't forget to use transpiler (if using modern syntax or operators) and polyfills (to add functions that may be missing). And they'll ensure that the code works.
+فراموش نکنید که از یک transpiler (اگر از syntax یا operatorهای مدرن استفاده می‌کنید) و polyfills (برای اضافه کردن فانکشن‌هایی که ممکن است موجود نباشند) استفاده کنید. و با این کار مطمئن خواهید بود که کد شما کار می‌کند.
 
-For example, later when you're familiar with JavaScript, you can setup a code build system based on [webpack](http://webpack.github.io/) with [babel-loader](https://github.com/babel/babel-loader) plugin.
+برای مثال، بعدها که با جاوااسکریپت آشنایی بیشتری پیدا کنید، می‌توانید یک سیستم build کد با [webpack](http://webpack.github.io/) و پلاگین [babel-loader](https://github.com/babel/babel-loader) بالا بیاورید.
 
-Good resources that show the current state of support for various features:
-- <https://kangax.github.io/compat-table/es6/> - for pure JavaScript.
-- <https://caniuse.com/> - for browser-related functions.
+منابع خوبی که نشان می‌دهند فیچرهای مختلف در چه حالتی از پشتیبانی قرار دارند:
 
-P.S. Google Chrome is usually the most up-to-date with language features, try it if a tutorial demo fails. Most tutorial demos work with any modern browser though.
+- <https://kangax.github.io/compat-table/es6/> - برای جاوااسکریپت.
+- <https://caniuse.com/> - برای فانکشن‌های مربوط به مرورگر.
 
```

Comment:
Do not add extra new lines.

Reply:
This is not an extra new line. The original file has this as far as I can tell. but I can delete it if you want.